src/app/api/services/dot-workflow-event-handler/dot-workflow-event-handler.service.ts

```diff
@@ -0,0 +1,215 @@
+                        this.openWizard(event);
+                    }
+                });
+        } else {
+            this.openWizard(event);
+        }
+    }
+
+    /**
+     * Check if there are environments present otherwise send a notification
+     * @memberof DotWorkflowEventHandlerService
+     */
+    checkPublishEnvironments(): Observable<boolean> {
+        return this.pushPublishService.getEnvironments().pipe(
+            take(1),
+            map((environments: DotEnvironment[]) => {
+                if (environments.length) {
+                    return true;
+                } else {
+                    this.dotMessageDisplayService.push({
+                        life: 3000,
+                        message: this.dotMessageService.get(
+                            'editpage.actions.fire.error.add.environment'
+                        ),
+                        severity: DotMessageSeverity.ERROR,
+                        type: DotMessageType.SIMPLE_MESSAGE
+                    });
+                }
+                return false;
+            })
+        );
+    }
+
+    /**
+     * Check if Push Publish is par of th sub-actions of the workflow.
+     * @param {DotCMSWorkflowInput[]} inputs
+     * returns boolean
+     * @memberof DotWorkflowEventHandlerService
+     */
+    containsPushPublish(inputs: DotCMSWorkflowInput[]): boolean {
+        return inputs.some(input => input.id === 'pushPublish');
+    }
+
+    /**
+     * Returns the input needed to collect information of the Workflow inputs.
+     *
+     * @param {DotCMSWorkflowAction} workflow
+     * @param {string} title
+     * @returns DotWizardInput
+     * @memberof DotWorkflowEventHandlerService
+     */
+    setWizardInput(workflow: DotCMSWorkflowAction, title: string): DotWizardInput {
+        const steps: DotWizardStep<any>[] = [];
+        this.mergeCommentAndAssign(workflow).forEach((input: DotCMSWorkflowInput) => {
+            if (this.workflowStepMap[input.id]) {
+                steps.push({
+                    component: this.workflowStepMap[input.id],
+                    data: input.body
+                });
+            }
+        });
+        return steps.length
+            ? {
+                  title: title,
+                  steps: steps
+              }
+            : null;
+    }
+
+    /**
+     * conver the data collected to what is expecting the endpoint.
+     * @param {{ [key: string]: any }} data
+     * @param {DotCMSWorkflowInput[]} inputs
+     * @returns { [key: string]: any }
+     * @memberof DotWorkflowEventHandlerService
+     */
+    processWorkflowPayload(
+        data: { [key: string]: any },
+        inputs: DotCMSWorkflowInput[]
+    ): { [key: string]: any } {
+        if (this.containsPushPublish(inputs)) {
+            data['whereToSend'] = data.environment.join();
+            data['iWantTo'] = data.pushActionSelected;
+            data['publishTime'] = moment(data.publishDate).format('HH-mm');
```

Comment:
check if this throws a warning in console due to it's deprecated, if it does then the fix is
`moment(new Date(data.publishDate)).format('HH-mm');`

Reply:
no.